### ISSUE TITLE
Queue.take() throws PollException

### ISSUE THREAD

**Issue by sahil-goel** [hazelcast/hazelcast#11868]
Hi ,

I have 5 hazelcast clients all waiting on different queues through queue.take()

Suddenly all of them threw an exception

```
No response for 120000 ms. Aborting invocation! Invocation{ serviceName='hz:impl:queueService', op=com.hazelcast.collection.impl.queue.operations.PollOperation{identityHash=1321665151, serviceName='null', partitionId=69, callId=3450443, invocationTime=1511478006227, waitTimeout=-1, callTimeout=60000}, partitionId=69, replicaIndex=0, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeout=60000, target=Address[<MEMBER HOST NAME>]:7500, backupsExpected=0, backupsCompleted=0} No response has been received!  backups-expected:0 backups-completed: 0
```

I am using Hazelcast 3.5.5.

I can also see following at member side 
```
com.hazelcast.core.OperationTimeoutException: No response for 120000 ms. Aborting invocation! Invocation{ serviceName='hz:impl:atomicLongService', op=com.hazelcast.concurrent.atomiclong.operations.AddAndGetOperation{identityHash=2061771286, serviceName='null', partitionId=101, callId=4144042, invocationTime=1511478018096, waitTimeout=-1, callTimeout=60000}, partitionId=101, replicaIndex=0, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeout=60000, target=Address[<Host NAME>]:7500, backupsExpected=0, backupsCompleted=0} No response has been received!  backups-expected:0 backups-completed: 0
```

**Comment by pveentjer** [comment]
Can you upgrade to HZ 3.9. There are a lot of improvements:
- a lot better OperationTimeoutExceptions that reveal more of what is happening (including the potential cause)
- a more reliable mechanism to detect these timeouts.

**Comment by sahil-goel** [comment]
Sorry but I can'y upgrade right away. It has been running in our production environment and if we have to upgrade, bets I could do is 3.8.8.

But for now, I really want to understand the reason behind above errors. It looks like, that exception came on member and it some how propagated to all clients that were waiting on hazelcast queue

**Comment by mmedenjak** [comment]
The operation timeout happens because the member executing the operation might have crashed, is stalled or has been disconnected from the rest of the cluster. The two error messages that you have posted are for two different actions: `IQueue#poll` and the other for one of `IAtomicLong` methods. You should check the logs for the operation targets (`target=Address[<MEMBER HOST NAME>]:7500` and `target=Address[<Host NAME>]:7500`) at the time of these messages to determine why that member has not returned a response.

**Comment by pveentjer** [comment]
In 3.6 it can also happen that when the system is just a bit slow, the 'is still running' operations don't get processed fast enough and you run into a false OperationTimeoutException (there is just a 6 second window in which the is-still-running request needs to get processed). This has been improved in 3.7 and has become less fragile.

**Comment by mmedenjak** [comment]
@sahil-goel as we've concluded, I don't see this as an issue - the operations simply timed out. In hazelcast 3.9 (and hazelcast 3.8.8) we have had significant improvements in the design. Please upgrade to the latest possible version. If this answer has not been satisfactory, you have more information or are experiencing a different problem, please reopen this one or open a new issue.
